src/lib.rs

```diff
@@ -474,6 +493,156 @@ impl XenControl {
         last_error!(self, max_gpfn)
     }
 
+    pub fn vcpu_getinfo(&self, domid: u32, vcpu: u32) -> Result<xc_vcpuinfo_t, XcError> {
```

Comment:
Returning a reexported `xc_vcpuinfo_t` works, but I notice that as an API consumer, that type isn't resolved by the IDE:
![image](https://github.com/Wenzel/xenctrl/assets/964610/37aeed59-d6c9-47b3-9c5a-c6b0de717ec3)

Rust knows then at compile time of course:
![image](https://github.com/Wenzel/xenctrl/assets/964610/c4fc7f19-0409-46e3-8a18-b60976196648)

Question is: should we make this API from dev friendly by wrapping the `xc_vcpuinfo_t` struct into a real Rust  struct ?

The struct as defined in `xen/domctl.h`
```C
/* XEN_DOMCTL_getvcpuinfo */
struct xen_domctl_getvcpuinfo {
    /* IN variables. */
    uint32_t vcpu;
    /* OUT variables. */
    uint8_t  online;                  /* currently online (not hotplugged)? */
    uint8_t  blocked;                 /* blocked waiting for an event? */
    uint8_t  running;                 /* currently scheduled on its CPU? */
    uint64_aligned_t cpu_time;        /* total cpu time consumed (ns) */
    uint32_t cpu;                     /* current mapping   */
};
```

What about defining an `XcVcpuInfo` type:

```rust
pub struct XcVcpuInfo {
    vcpu: u32,
    online: u8,
    blocked: u8,
    running: u8,
    cpu_time: u64,
    cpu: u32,
}
```

And implementing the `From` trait to make the conversion automatically

```rust
impl From<xc_vcpuinfo_t> for XcVcpuInfo {
    fn from(value: xc_vcpuinfo_t) -> Self {
        Self {
            vcpu: value.vcpu,
            online: value.online,
            blocked: value.blocked,
            running: value.running,
            cpu_time: value.cpu_time,
            cpu: value.cpu,
        }
    }
}
```
last line could be rewritten as:

```rust
last_error!(self, XcVcpuInfo::from(vcpu_info))
```